### DISCUSSION TITLE
2.3.40 to 2.3.50-70 Multi Fields

### DISCUSSION THREAD

**Discussion by 8inaryMata1eao** [Security-Onion-Solutions/securityonion#5538] [Unsupported Versions] [answered]
I have noticed an issue since upgrading from 2.3.50 that has not yet resolved.  I thought it might own its own but after a while its still a constant problem.  Many of the fields in kibana show "Multi fields" and duplicate date with .keyword and .security.  How do I resolve this?

![image](https://user-images.githubusercontent.com/80425093/133646589-bd00c60e-17bb-4568-8d22-8b783f47ec8c.png)

![image](https://user-images.githubusercontent.com/80425093/133646784-499be8f6-63a9-467c-b78d-5a871482617b.png)

**Comment by dougburks** [comment, marked as answer]
I think what you are seeing is normal for newer versions of Kibana.

**Reply by 8inaryMata1eao** [reply]
Doug thanks for the info.  So is this the expected behavior?  How can I remove the duplicate fields? It definitely makes analysis tricky with all of the extra lines.

**Reply by dougburks** [reply]
Google suggests a potential workaround:
https://github.com/elastic/kibana/issues/96029#issuecomment-819658319

You could also try using our Hunt interface:
http://securityonion.net/docs/hunt

**Reply by rwaight** [reply]
Hey @8inaryMata1eao and @dougburks, this is mainly due to the `"dynamic": true` settings within the [so-common-template](https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/templates/so/so-common-template.json), it is currently configured to allow for [multi-field](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/multi-fields.html#_multi_fields_with_multiple_analyzers) with the standard and [security](https://docs.securityonion.net/en/2.3/playbook.html?highlight=analyzer#security-subfield) analyzers.  In order to limit/prevent multi-fields, the mapping would have to be updated to specify how the field type.  

For example, I tried to specify `event.module` as a `keyword` in the mapping, but found that both the **hunt** application (looks for the `event.module.security` subfield) and the default SO Kibana dashboard (look for the `event.module.keyword` subfield) stopped working with the mapping changes I made.  If I wanted to set `event.module` to a keyword (which [aligns with ECS](https://www.elastic.co/guide/en/ecs/1.10/ecs-event.html#field-event-module)) and still allow the **hunt** application to work, I would need to update the mapping accordingly.  On that note, it might make sense to update mappings to reduce the number of fields that have multi-field enabled, but it appears that would cause a breaking change with other parts of Security Onion.

If you tried to hide the extra fields, it would only be cosmetic as there would still be more data stored within a single document.  The reason I mention that is hiding the fields does not reduce the exponential amount of multi-fields that would exist in an environment with billions and trillions of events/logs (i.e., every document has hundreds of fields with multi-field enabled multiplied by the total number of documents in the cluster).

**Reply by dougburks** [reply]
@rwaight Thanks for the detail. Tagging @weslambert for visibility as he is working on improving our templates and ECS support.
